src/Table.tsx

```diff
@@ -24,6 +26,10 @@ interface Props<RowShape> {
      * an id for the column
      */
     id: string | number;
+    /**
+     * the width for the column
+     */
+    width?: string | number;
```

Comment:
I think we can come up with a better long-term solution here. After reading [Optimized for Change](https://overreacted.io/optimized-for-change/) I've tried to think of solutions in terms of how easy to maintain they'll be in the future. Here we'd be adding a custom prop for the `column` config. What if we want to add other styles? What if we want to add other props? How can we solve those problems right now?

I think a better solution would take from the solution that @cheapsteak came up with in engine-frontend:

https://github.com/mdg-private/engine-frontend/blob/527b6b9d644fe7150e73e6a1f49c1516048545fb/imports/ui/components/TableWithDrawers.tsx#L37-L41

I think we should create an option in `column` for that will take either an object or a function that accepts `RowData` and returns an object that will be applied directly to the `td` or `th` for the column. Then we can add `cellProps={{ className: "w-16" }}` the column config (or an emotion variant for projects outside of engine-frontend) and that can be spread directly to the `th`s and `td`s. Then if we ever need other props to be added, the logic is already there.

What do you think @fmvfelipemonsalve ?

Reply:
Proposed update:

```ts
/**
 * Add props directly to this column's `td`
 *
 * Accepts either an object or a function taking the row data and returning
 * an object. The result will be spread directly to the `td` with the
 * `className` being merged in.
 */
tableDataProps?:
  | React.DetailedHTMLProps<
      React.HTMLAttributes<HTMLTableRowElement>,
      HTMLTableRowElement
    >
  | ((
      rowShape: RowShape
    ) => React.DetailedHTMLProps<
      React.HTMLAttributes<HTMLTableRowElement>,
      HTMLTableRowElement
    >);

/**
 * Add props to the header's `<th>`
 *
 * This is useful for configuring the size of the column
 */
tableHeaderProps?: React.DetailedHTMLProps<
  React.ThHTMLAttributes<HTMLTableHeaderCellElement>,
  HTMLTableHeaderCellElement
>;
```

Reply:
Another option, perhaps even simpler, would be to use `<colgroup>` This also would need to be configured into the table, but would be done similarly:

https://www.w3schools.com/tags/tag_colgroup.asp

```tsx
/**
 * Pass these props to `col` elements nested below the `table`'s single
 * `<colgroup>`.
 *
 * This allows you to apply styles to columns (including setting sizes) by
 * setting a class on a single element instead of _all_ elements in a
 * table's row
 */
colProps?: React.DetailedHTMLProps<
  React.ColHTMLAttributes<HTMLTableColElement>,
  HTMLTableColElement
>;
```

Reply:
After discussing with Justin, the colgroup option seems like a good way to go. Less code and you can pass span, or width etc as a consumer.